overlay/overlay-shared.pro

```diff
@@ -36,17 +35,24 @@ CONFIG(force-x86_64-toolchain) {
   LIBS *= -lminhook
 }
 
-CONFIG(release, debug|release) {
-  DESTDIR = ../release
-  QMAKE_LIBDIR = ../release $$QMAKE_LIBDIR
-}
-
 CONFIG(debug, debug|release) {
-  DESTDIR = ../debug
-  QMAKE_LIBDIR = ../debug $$QMAKE_LIBDIR
   DEFINES *= DEBUG
 }
 
+CONFIG(custom_build_dir) {
+  OBJECTS_DIR = $$OBJECTS_DIR/overlay/$$TARGET
```

Comment:
This works, even when TARGET is defined later on? I still don't have a good mental model for how qmake works, I'm afraid 😅

Reply:
Yes, that's why I didn't have to move `include(../qmake/compiler.pri)` in the plugins' project file.